### DISCUSSION TITLE
What is your general workflow when creating your graphics?

### DISCUSSION THREAD

**Discussion by larrybud2004** [Bodmer/TFT_eSPI#3039] [General]
I'm using this library to help create informational screens and interactive option selections for a personal fun golf app (no animations at this time), and I'm wondering how others are handling their workflow:

- Do you generate your screens outside any development area (i.e. photoshop/illustrator) first, and mimic them using eSPI?
- Or do you just create your screens on the fly, not knowing what size text, font type, etc and tweak it later?
- Or something else that I might not have thought of?

Currently I'm just doing a seat of my pants development making sure functionality in the app is working, and thinking that I will improve on the graphics later, but I don't know if that is the most efficient use of my time.

Any other advice is welcome.

**Comment by MYCAMEL222** [comment]
Try Canva.....They have a nice free version, but I pay for the pro version ($12 month)

You can make small sprites or full screens. Use the PNGdec.h library to decode the PNG into an array. 

I use a fadein/fadeout function (fade the backlight LED) so the user doesn't see the screens being written to the TFT.

It's just how I do it. 

![LOAD CELL TESTER (3)](https://github.com/Bodmer/TFT_eSPI/assets/85456085/6cb52d44-7b96-4734-9e5d-51c15bdebb7f)

**Comment by larrybud2004** [comment]
Thanks. So all your menus are pure graphics? Do you do any kind of "button press" highlights? Would it be possible to attach a video showing some of the things you've done?

**Comment by MYCAMEL222** [comment]
Here you go. Not HQ due to files size limitations. I don't worry about the button "highlights". Just give them all the options on the screen and they'll press the right button...

https://github.com/Bodmer/TFT_eSPI/assets/85456085/402985e2-bfaa-4065-90f3-3826231a595b

**Comment by larrybud2004** [comment]
Excellent! Thanks for the feedback. Your screens look great.

**Reply by MYCAMEL222** [reply]
Drawing every button and graphic in TFT_eSPI is possible, but I soon learned that drawing in a graphics program and then importing the PNG via an array into the sketch was a simpler approach. 

TFT_eSPI is a great library. 

Well done Bodmer!!!
